server/internal/nvdp/watchers.go

```diff
@@ -0,0 +1,101 @@
+package nvdp
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost/server/public/model"
+
+	"github.com/illbjorn/mm-nvd/nvd"
+)
+
+// WatcherSubscribe initiates the process of announcing CVEs to the channel the
+// command was invoked in.
+func (p *Plugin) WatcherSubscribe(args *model.CommandArgs) {
+	// don't register the same channel twice
+	if p.cveWG.Worker(args.ChannelId) == nil {
+		// register a callback to fire a message to this channel when matching
+		// criteria CVEs are identified
+		p.cveWG.Register(
+			args.ChannelId,
+			nvd.NewCVEQuery(),
+			func(cve nvd.CVE) {
+				// create the post
+				post := p.NewCVEPost(cve, args.ChannelId, args.RootId)
+				// ship it!
+				_, _ = p.API.CreatePost(post)
+			})
+
+		// notify of subscription success
+		// TODO: this should be an ephemeral message to the user that invoked the
+		//       command.
+		post := p.NewPost("Subscription successful!", args.ChannelId, args.RootId)
+		_, _ = p.API.CreatePost(post)
+	}
+}
+
+// WatcherUnsubscribe unsubscribes (discontinues CVE announcements) for the
+// channel the command was invoked in.
+func (p *Plugin) WatcherUnsubscribe(args *model.CommandArgs) {
+	// only attempt unsubscribe if we've actually registered a worker for this
+	// channel
+	if worker := p.cveWG.Worker(args.ChannelId); worker != nil {
+		// perform the unregister
+		if err := p.cveWG.Unregister(args.ChannelId); err == nil {
+			// TODO: really should be doing something with these errors.
```

Comment:
What would you do?

Reply:
I left these comments more so to indicate that I know something _should_ be done with them but I didn't understand well enough how Mattermost's error handling and logging works (ex: is there a package in the Mattermost Go packages that can log to the System Console?) to implement something here.

Just in general: I'd be logging them, shipping them to a log aggregation stack, things like these.